src/NJsonSchema/Generation/JsonSchemaGenerator.cs

```diff
@@ -924,7 +930,7 @@ private async Task LoadPropertyOrFieldAsync(Newtonsoft.Json.Serialization.JsonPr
                 var isNullable = propertyTypeDescription.IsNullable &&
                     hasRequiredAttribute == false &&
                     (bool)isDataContractMemberRequired == false &&
-                    (property.Required == Required.Default || property.Required == Required.AllowNull);
+                    (propertyRequired == Required.Default || propertyRequired == Required.AllowNull);
```

Comment:
Be careful here:
If I set `[JsonObject(ItemRequired = Required.Always)]`, will I still be able to write `public DateTime? ExpirationDate { get; set; }` without any attribute, and having it nullable ? This is an issue I have with my workaround, and I need `[CanBeNull]` for now.